### ISSUE TITLE
ML notebook: Add preprocessing and a sklearn pipeline

### ISSUE THREAD

**Issue by rhiever** [rhiever/Data-Analysis-and-Machine-Learning-Projects#7]
From a [Reddit comment](https://www.reddit.com/r/Python/comments/3hu8fj/im_creating_an_example_python_machine_learning/cuaqt9g):

>Advice: I think you are missing a few big things like preprocessing/scaling and pipelines.

>Before using the learners, inputs should be scaled so that each feature has equal weight. Something like StandardScaler or MinMaxScaler are both appropriate (from sklearn.preprocessing). If you think some features are more important, you can scale them later to increase their relative importance in prediction. These are more parameters you would tune using CV, but these can be really numerous, so GridSearch is out the window and you would have to consider some alternatives like Nelder Mead search, genetic search, or multivariate gradient descent if you suspect convexity.

>You have to fit these scalers on the training data and then use the trained fit to transform the testing data. Using Pipelines simplifies this whole process (fits the scaler and learner at once, transforms and predicts at once).

**Comment by rhiever** [comment]
Comment from Sebastian fits in nicely with this change:

>Maybe mention that random forests are scale-invariant, e.g., you could mention that a typical procedure in the data preprocessing pipeline (required by most ML algos) is to scale the features because you are using decision trees (I believe this is the only scale-invariant algo that is used in ML) -- maybe also explain what a decision tree is and how it relates to random forests in a few sentences. On a side-note, but you probably already now this: Most gradient-based optimization algos

**Comment by Ritish-Madan** [comment]
> From a [Reddit comment](https://www.reddit.com/r/Python/comments/3hu8fj/im_creating_an_example_python_machine_learning/cuaqt9g):
> 
> > Advice: I think you are missing a few big things like preprocessing/scaling and pipelines.
> > Before using the learners, inputs should be scaled so that each feature has equal weight. Something like StandardScaler or MinMaxScaler are both appropriate (from sklearn.preprocessing). If you think some features are more important, you can scale them later to increase their relative importance in prediction. These are more parameters you would tune using CV, but these can be really numerous, so GridSearch is out the window and you would have to consider some alternatives like Nelder Mead search, genetic search, or multivariate gradient descent if you suspect convexity.
> > You have to fit these scalers on the training data and then use the trained fit to transform the testing data. Using Pipelines simplifies this whole process (fits the scaler and learner at once, transforms and predicts at once).

Can you please tell which file you are referring to? To add the scaling.

**Comment by rhiever** [comment]
The example data science notebook: https://github.com/rhiever/Data-Analysis-and-Machine-Learning-Projects/blob/master/example-data-science-notebook/Example%20Machine%20Learning%20Notebook.ipynb

**Comment by Ritish-Madan** [comment]
Hi!I believe that Scaling the features won't really be beneficial in our features. The classification is based on the size of petals and various other continuous data features are available. Scaling them might ruin their actual existence and might ruin the classification.
